core/src/test/java/org/bptlab/cepta/count-trains-in-station-tests/CountTrainsInStationTests.java

```diff
@@ -0,0 +1,125 @@
+        LiveTrainDataOuterClass.LiveTrainData end =
+                LiveTrainDataProvider.getDefaultLiveTrainDataEvent().toBuilder()
+                        .setStationId(commonStation)
+                        .setTrainId(3)
+                        .setEventTime(Timestamps.parse("2042-01-01T10:00:31.023-05:00")).build();
+
+        DataStream<LiveTrainDataOuterClass.LiveTrainData> liveTrainStream =
+                env.fromElements(first, second, end)
+                .assignTimestampsAndWatermarks(StreamUtils.eventTimeExtractor());
+
+        DataStream<CountOfTrainsAtStationEvent> countOfStationStream = CountOfTrainsAtStationFunction.countOfTrainsAtStation(liveTrainStream);
+        ArrayList<CountOfTrainsAtStationEvent> resultCollection = StreamUtils.collectStreamToArrayList(countOfStationStream);
+
+        /*Expect 4 times cause the function provides a sliding window of one hour every 15 minutes*/
+        Assert.assertEquals(4, resultCollection.size());
+    }
+
+    @Test
+    public void TestCorrectCountAtStation() throws IOException, ParseException {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(1);
+        int commonStation = 1;
+        LiveTrainDataOuterClass.LiveTrainData first =
+                LiveTrainDataProvider.getDefaultLiveTrainDataEvent().toBuilder()
+                        .setStationId(commonStation)
+                        .setTrainId(1)
+                        .setEventTime(Timestamps.parse("1972-01-01T10:00:20.021-05:00")).build();
+        LiveTrainDataOuterClass.LiveTrainData second =
+                LiveTrainDataProvider.getDefaultLiveTrainDataEvent().toBuilder()
+                        .setStationId(commonStation)
+                        .setTrainId(2)
+                        .setEventTime(Timestamps.parse("1972-01-01T10:00:21.022-05:00")).build();
+        LiveTrainDataOuterClass.LiveTrainData end =
+                LiveTrainDataProvider.getDefaultLiveTrainDataEvent().toBuilder()
+                        .setStationId(commonStation)
+                        .setTrainId(3)
+                        .setEventTime(Timestamps.parse("1972-01-02T10:00:21.022-05:00")).build();
+
+        DataStream<LiveTrainDataOuterClass.LiveTrainData> liveTrainStream =
+                env.fromElements(first, second, end)
+                .assignTimestampsAndWatermarks(StreamUtils.eventTimeExtractor());
+
+
+        DataStream<CountOfTrainsAtStationEvent> countOfStationStream = CountOfTrainsAtStationFunction.countOfTrainsAtStation(liveTrainStream);
+        ArrayList<CountOfTrainsAtStationEvent> resultCollection = StreamUtils.collectStreamToArrayList(countOfStationStream);
+
+        for (CountOfTrainsAtStationEvent in : resultCollection) {
+                Assert.assertEquals(1, in.getStationId());
+                Assert.assertEquals(2, in.getCount());
```

Comment:
I think it would be nicer to create a referecne/expected array and compare the resultCollection to that. That way you can test for the whole objects

Reply:
The problem here is that the window size + sliding interval determines how many events we get. So we would need to adjust the tests every time we adjust those two values.
This way we just say that each event needs to have the correct count and station and we test the amount somewhere else. 

I guess we could change it tho, what do you think?